provider/src/main/kotlin/org/gradle/kotlin/dsl/provider/KotlinBuildScriptCompiler.kt

```diff
@@ -232,6 +229,22 @@ class KotlinBuildScriptCompiler(
             script,
             classPath)
 
+    private
+    fun withClassFrom(
+        compiledScript: CachingKotlinCompiler.CompiledScript,
+        classLoaderScope: ClassLoaderScope,
+        action: (Class<*>) -> Unit) {
+
+        val scriptClass = classFrom(compiledScript, classLoaderScope)
+        withContextClassLoader(scriptClass.classLoader) {
+            try {
+                action(scriptClass)
+            } catch (e: InvocationTargetException) {
+                throw e.targetException
```

Comment:
Why are you dropping the outer `InvocationTargetException`?

Reply:
Because it adds no value.

Reply:
Okay. Fair enough. Just as long as no useful information is lost.